test/unit/models/inbound/accounts.model.test.ts

```diff
@@ -0,0 +1,199 @@
+        userId,
+        {
+          errorInformation: {
+            errorCode: '2003',
+            errorDescription: 'Service currently unavailable'
+          }
+        },
+        dfspId
+      )
+    })
+
+    test('reformating of thrown exception when res.data present and using different statusCode', async () => {
+      mocked(config.thirdpartyRequests.putAccounts).mockImplementationOnce(
+        () => {
+          throw new HTTPResponseError({
+            msg: 'mocked-error',
+            res: {
+              data: { statusCode: '2002' }
+            }
+          })
+        }
+      )
+
+      await model.getUserAccounts(userId, dfspId)
+
+      expect(config.backendRequests.getUserAccounts).toHaveBeenCalledWith(userId)
+      expect(config.thirdpartyRequests.putAccountsError).toHaveBeenCalledWith(
+        userId,
+        {
+          errorInformation: {
+            errorCode: '2002',
+            errorDescription: 'Not implemented'
+          }
+        },
+        dfspId
+      )
+    })
+
+    test('reformating of thrown generic Error', async () => {
+      mocked(config.thirdpartyRequests.putAccounts).mockImplementationOnce(
+        () => {
+          throw new Error('generic-error')
+        }
+      )
+
+      await model.getUserAccounts(userId, dfspId)
+
+      expect(config.backendRequests.getUserAccounts).toHaveBeenCalledWith(userId)
+      expect(config.thirdpartyRequests.putAccountsError).toHaveBeenCalledWith(
+        userId,
+        {
+          errorInformation: {
+            errorCode: '2001',
+            errorDescription: 'Internal server error'
+          }
+        },
+        dfspId
+      )
+    })
+
+    test('reformating of thrown exception when res.body is not valid JSON string', async () => {
+      mocked(config.thirdpartyRequests.putAccounts).mockImplementationOnce(
+        () => {
+          throw new HTTPResponseError({
+            msg: 'mocked-error',
+            res: {
+              body: { statusCode: '2002' }
+            }
+          })
+        }
+      )
+
+      await model.getUserAccounts(userId, dfspId)
+
+      expect(config.backendRequests.getUserAccounts).toHaveBeenCalledWith(userId)
+      expect(config.thirdpartyRequests.putAccountsError).toHaveBeenCalledWith(
+        userId,
+        {
+          errorInformation: {
+            errorCode: '2001',
+            errorDescription: 'Internal server error'
+          }
+        },
+        dfspId
+      )
+    })
+
+    test('reformating of thrown exception when no user accounts returned', async () => {
+      mocked(config.backendRequests.getUserAccounts).mockImplementationOnce(() => Promise.resolve())
+
+      await model.getUserAccounts(userId, dfspId)
+
+      expect(config.backendRequests.getUserAccounts).toHaveBeenCalledWith(userId)
+      expect(config.thirdpartyRequests.putAccountsError).toHaveBeenCalledWith(
+        userId,
+        {
+          errorInformation: {
```

Comment:
shouldn't it be a generic ID not-found error? 3200

Reply:
Fixed,  previously checking empty response, but ideally simulator returns 3200 error if user id not found